public/audioplayer.html

```diff
@@ -65,4 +71,13 @@
         })
     }
 
+    function playScale() {
+        let context = new (window.AudioContext || window.webkitAudioContext)();
+        let note = new Sound(context);
+        let now = context.currentTime;
+        queryObject.scaleNotes.forEach(array => {
+            note.play(array[0], now + array[1], now + array[2]);
+        })
```

Comment:
you pull pull all this into a function that takes the notes to play as a param
```js
function playNotes(notes) {
   let context = new (window.AudioContext || window.webkitAudioContext)();
   let note = new Sound(context);
   let now = context.currentTime;
   notes.forEach(array => {   
     note.play(array[0], now + array[1], now + array[2]);
   })
}
```

and then each specific function can know how to get the right notes and pass it along
```js
function playScale() {
  playNotes(queryObject.scaleNotes)
}

function playMelody() {
  playNotes(queryObject.melody)
}
```

Reply:
yess thanks!